addons/mail/static/src/core/common/attachment_model.js

```diff
@@ -21,6 +21,12 @@ export class Attachment extends FileModelMixin(Record) {
     static insert(data) {
         return super.insert(...arguments);
     }
+    static _insert() {
+        /** @type {import("models").Attachment} */
+        const attachment = super._insert(...arguments);
+        attachment.message ??= attachment.message_ids?.[0];
+        return attachment;
+    }
```

Comment:
This should not be done in `_insert` (we never want to override that), it should just define the field `message_ids` and then a getter or a computed field for `message`.

Reply:
And ideally if it's not too complicated instead of taking `message_ids[0]` it should take the (first?) message that is in the same thread (in case we do really have an attachment linked to several messages).

Reply:
For the second part, we would need to know the thread each message in `message_ids` is in, which we currently do not.
We could change the to_store to send filtered `message_ids` that are in the `thread` also sent.